src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -156,10 +157,27 @@ private RuleSet FindDeclarationRuleSet(Project project, RuleSetDeclaration decla
             return ruleSetSerializer.LoadRuleSet(ruleSetFilePath);
         }
 
-        private bool ProjectHasAdditionalFile(Project project, string additionalFilePath)
+        public bool DoesProjectReferenceAdditionalFile(Project project, string solutionAdditionalFilePath)
         {
-            // ritag todo: 1. Check if there is another SonarLint.xml, 2. Check if the property is AdditionalFiles
-            return projectSystemHelper.IsFileInProject(project, additionalFilePath);
+            var additionalFile = projectSystemHelper.FindFileInProject(project, solutionAdditionalFilePath);
+
+            if (additionalFile == null)
+            {
+                var additionalFileName = Path.GetFileName(solutionAdditionalFilePath);
+                var hasAnotherAdditionalFile = projectSystemHelper.IsFileInProject(project, additionalFileName);
```

Comment:
FYI I couldn't get VS to hit this code path when debugging - "IsFileInProject" never returned `true` when searching by just the file name.